engine/src/test/java/org/camunda/bpm/engine/test/api/runtime/RestartProcessInstanceAsyncTest.java

```diff
@@ -1150,6 +1151,32 @@ public void shouldSetExecutionStartTimeInBatchAndHistory() {
     Assertions.assertThat(historicBatch.getExecutionStartTime()).isEqualToIgnoringMillis(TEST_DATE);
   }
 
+  @Test
+  public void shouldSetProcessInstanceAndDefinitionIdInHistoryJobLog() {
+    // given
+
+    ProcessDefinition processDefinition = testRule.deployAndGetDefinition(ProcessModels.TWO_TASKS_PROCESS);
+    ProcessInstance processInstance = runtimeService.createProcessInstanceById(processDefinition.getId())
+            .startBeforeActivity("userTask1")
+            .execute();
+    runtimeService.deleteProcessInstance(processInstance.getId(), "test");
+    Batch batch = runtimeService.restartProcessInstances(processDefinition.getId())
+            .startAfterActivity("userTask2")
+            .processInstanceIds(processInstance.getId())
+            .executeAsync();
+    helper.executeSeedJob(batch);
+    List<Job> executionJobs = helper.getExecutionJobs(batch);
+
```

Comment:
✍️ After executing the seed job and before executing the batch execution jobs you can perform the check if `executionJobs.get(0)` has assigned correctly the process instance and definition ids.

Reply:
Thank you @yanavasileva . I should have thought about that. I've made the change